### PR TITLE
Issues in health insight support

### PR DESCRIPTION
**Pull request by qiaozha**

### CODE REVIEW

File: packages/typespec-ts/src/modelUtils.ts [hunk 1]
```diff
@@ -453,6 +453,8 @@ function getSchemaForModel(
     true /** shouldGuard */
   );
 
+  modelSchema.typeName = modelSchema.name;
```
Comment (qiaozha) [inline packages/typespec-ts/src/modelUtils.ts:456]: This is the only change. not sure why Health Insight fail to generate the propertyType name if that property is a Model, I tried to repro it in the unit-test, still unable to repo.

Comment (MaryGao) [reply packages/typespec-ts/src/modelUtils.ts:456]: All models in property failed or just some specific one?

Comment (qiaozha) [reply packages/typespec-ts/src/modelUtils.ts:456]: all.

**Review by MaryGao — APPROVED** [review]
LGTM

File: packages/typespec-ts/src/modelUtils.ts (hunk 1 above)
Comment (MaryGao) [reply packages/typespec-ts/src/modelUtils.ts:456]: Approved but still prefer to find the root cause to repro in UT.

**Pull request merged**